File: src/schemas/queries.py
from pydantic import BaseModel


class SearchQuery(BaseModel):
    search_string:      str|None
    actor:              str|None
    studio:             str|None
    collection:         str|None
    include_terms:      list[str]
    exclude_terms:      list[str]
    date_added_from:        str|None
    date_added_to:          str|None
    date_released_from:     str|None
    date_released_to:       str|None
    only_favourites:    bool
    sortby:             str|None
    limit:              int
    startfrom:          int


class CatalogueQuery(BaseModel):
    query_type:          str  # performers | studios
    query_string:        str|None = None
    use_primary_actors:  bool = False
    
    filter_actor:       str|None = None
    filter_studio:      str|None = None
    filter_collection:  str|None = None
    filter_tag:         str|None = None

File: go_backend/internal/schemas/queries.go
package schemas

type SearchQuery struct {
	SearchString     string   `json:"search_string"`
	Actor            string   `json:"actor"`
	Studio           string   `json:"studio"`
	Collection       string   `json:"collection"`
	IncludeTerms     []string `json:"include_terms"`
	ExcludeTerms     []string `json:"exclude_terms"`
	DateAddedFrom    string   `json:"date_added_from"`
	DateAddedTo      string   `json:"date_added_to"`
	DateReleasedFrom string   `json:"date_released_from"`
	DateReleasedTo   string   `json:"date_released_to"`
	OnlyFavourites   string   `json:"only_favourites"`
	SortBy           string   `json:"sortby"`
	Limit            int      `json:"limit"`
	StartFrom        int      `json:"startfrom"`
}

type CatalogueQuery struct {
	QueryType        string `json:"query_type"` // actors | studios
	QueryString      string `json:"query_string"`
	UsePrimaryActors bool   `json:"use_primary_actors"`

	FilterActor      string `json:"filter_actor"`
	FilterStudio     string `json:"filter_studio"`
	FilterCollection string `json:"filter_collection"`
	FilterTag        string `json:"filter_tag"`
}
